server/utils/taskQueues.js

```diff
@@ -11,14 +11,25 @@ async function setScraperInterval(cron) {
       doc != null;
       doc = await cursor.next()
     ) {
-      await scrapingAmazon(doc.url).then(function (updated) {
-        if (updated != null) {
-          doc.name = updated.title;
-          doc.imageUrl = updated.imageUrl;
-          doc.price = updated.price;
-        }
-        doc.save();
-      });
+      if (doc.url.includes("www.ebay.")) {
+        await scrapingEbay(doc.url).then(function (updated) {
+          if (updated != null) {
+            doc.name = updated.title;
+            doc.imageUrl = updated.imageUrl;
+            doc.price = updated.price;
+          }
+          doc.save();
+        });
+      } else if (doc.url.includes("www.amazon.")) {
+        await scrapingAmazon(doc.url).then(function (updated) {
+          if (updated != null) {
+            doc.name = updated.title;
+            doc.imageUrl = updated.imageUrl;
+            doc.price = updated.price;
+          }
+          doc.save();
+        });
+      }
```

Comment:
Looks like you're repeating code here, can you refactor this to avoid repeating any code? Maybe you can have a generic callback variable that gets assigned based on the url?